zaza/openstack/charm_tests/octavia/tests.py

```diff
@@ -453,3 +453,36 @@ def test_create_loadbalancer(self):
 
         # If we get here, it means the tests passed
         self.run_resource_cleanup = True
+
+class VolumeAmphoraTest(LBAASv2Test):
+    """LBaaSv2 Volume-based Amphora tests"""
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for LBaaSv2 Volume-based amphora tests
+           only for Openstack Train release and newer releases.
+         """
+        if (openstack_utils.get_os_release() <
+                openstack_utils.get_os_release('bionic_train')):
+            return
+        super(VolumeAmphoraTest, cls).setUpClass()
+
+    def test_volume_based_amphora(self):
+        default_charm_config = {'enable-volume-based-amphora': False}
+        alternate_charm_config = {'enable-volume-based-amphora': True}
+        with self.config_change(default_charm_config, alternate_charm_config,
+            reset_to_charm_default=True):
+            logging.info("Enabled volume based amphora setting.")
+            amphora_list = self.octavia_client.amphora_list()
+            self.assertTrue(len(amphora_list) > 0)
+            attached_volumes = []
+            for amphora in amphora_list.get('amphorae', []):
+                for server in self.nova_client.servers.list():
+                    if 'compute_id' in amphora and server.id == amphora[
+                         'compute_id']:
+                        server_id = amphora['compute_id']
+                        attached_volumes.append(json.dumps(vars(self.nova_client.volumes.get_server_volumes(server_id))))
+            self.assertTrue(len(attached_volumes) > 0)
+            logging.info("Amphora volumes creation successful: {}",
+                attached_volumes)
+    def test_create_loadbalancer(self):
+        super().test_create_loadbalancer()
```

Comment:
I'm guessing we need the super() call to make the test actually run (due to inheritance).  However, it needs a line above it otherwise I think that the pep8 check should fail.  Did you run `tox -e pep8`?

Reply:
Thanks. I'd made a rookie mistake not running the lint tests. The CI caught this